processor/src/test/java/org/mapstruct/ap/test/java8stream/StreamMappingTest.java

```diff
@@ -0,0 +1,244 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.java8stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+@WithClasses({
+    Source.class,
+    Target.class,
+    Colour.class,
+    SourceTargetMapper.class,
+    TestList.class,
+    StringHolderArrayList.class,
+    StringHolder.class
+})
+@IssueKey( "962" )
+@RunWith(AnnotationProcessorTestRunner.class)
+public class StreamMappingTest {
+
+    @Test
+    public void shouldMapNullList() {
+        Source source = new Source();
+
+        Target target = SourceTargetMapper.INSTANCE.sourceToTarget( source );
+
+        assertThat( target ).isNotNull();
+        assertThat( target.getStringList() ).isNull();
+    }
+
+    @Test
+    public void shouldReverseMapNullList() {
+        Target target = new Target();
+
+        Source source = SourceTargetMapper.INSTANCE.targetToSource( target );
+
+        assertThat( source ).isNotNull();
+        assertThat( source.getStringList() ).isNull();
+    }
+
+    @Test
+    public void shouldMapList() {
```

Comment:
>Why is it using the identity function here?

Identity is used so I can have a general approach. Will fix this. 

>And why is it assigning the result to a new list?

The assigning to a new list is due to the before/after mapping possible calls. I can try and fix it by adding some checks if there are any before/after calls. Maybe add it as a TODO in #1012?

>I think we could avoid the assignment of Stream.empty() which is overridden right in the next line? 

Same for the `Stream.empty()`.  There could be calls to before/after mapping methods (Although it makes no sense to me to have a before mapping for `Stream`). I can do something similar as for the list.

>And the call to map() looks not properly indented (I know, it's nit-picking, sorry ;).

I am having problems to indent this correctly :(. I am not sure how I can do it correctly, @sjaakd maybe you can help out for this one. It is in line [114](https://github.com/filiphr/mapstruct/blob/java8-stream-conversions/processor/src/main/resources/org/mapstruct/ap/internal/model/StreamMappingMethod.ftl#L114), [117](https://github.com/filiphr/mapstruct/blob/java8-stream-conversions/processor/src/main/resources/org/mapstruct/ap/internal/model/StreamMappingMethod.ftl#L117) and [120](https://github.com/filiphr/mapstruct/blob/java8-stream-conversions/processor/src/main/resources/org/mapstruct/ap/internal/model/StreamMappingMethod.ftl#L120)